model/registry/artifact.yaml

```diff
@@ -0,0 +1,60 @@
+groups:
+  - id: registry.artifact
+    prefix: artifact
+    type: attribute_group
+    brief: >
+     This group describes attributes specific to artifacts.
+    attributes:
+      - id: name
+        type: string
+        stability: experimental
+        brief: >
+          The human readable file name of the artifact, often generated by
+          build and release utilities to include the package name and version.
+        examples: ['golang-binary-amd64-v0.1.0', 'docker-image-amd64-v0.1.0', 'release-1.tar.gz', 'file-name-package.tar.gz']
+      - id: version
+        type: string
+        stability: experimental
+        brief: >
+          The version of the artifact.
+        examples: ['v0.1.0', '1.2.1', '122691-build']
+      - id: purl
+        type: string
+        stability: experimental
+        brief: >
+          The [Package URL](https://github.com/package-url/purl-spec) of the
+          artifact, providing a standard way to identify and locate the
+          artifact.
+        examples: ['pkg:github/package-url/purl-spec@1209109710924', 'pkg:npm/foo@12.12.3']
+      - id: hash
+        type: string
+        stability: experimental
+        brief: >
+          The full [hash value (see glossary)](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-5.pdf),
+          often found in checksum.txt on a release of the artifact and used to
+          verify pakage integrity.
+        examples: ['9ff4c52759e2c4ac70b7d517bc7fcdc1cda631ca0045271ddd1b192544f8a3e9']
+      - id: attestation.id
+        type: string
+        stability: experimental
+        brief: >
+          The id of the build [software attestation](https://slsa.dev/attestation-model).
+        examples: ['123']
+      - id: attestation.filename
```

Comment:
@lmolkova  - I could argue this could be just `name` instead of `filename`. The SLSA link in the brief calls out filename directly, which is why I chose it here. But we have an attribute of `name` above for the package which is the filename. Any thoughts?

Reply:
since I'm thinking about artifact as a package and it's name as a public name, I think filename for attestation makes sense.

E.g. 
https://repo1.maven.org/maven2/io/opentelemetry/opentelemetry-api/1.40.0/

```
opentelemetry-api-1.40.0-javadoc.jar              2024-07-05 22:26    296665      
opentelemetry-api-1.40.0-javadoc.jar.asc          2024-07-05 22:26       478      
opentelemetry-api-1.40.0-javadoc.jar.asc.md5      2024-07-05 22:26        32      
opentelemetry-api-1.40.0-javadoc.jar.asc.sha1     2024-07-05 22:26        40      
opentelemetry-api-1.40.0-javadoc.jar.asc.sha2...  2024-07-05 22:26        64      
opentelemetry-api-1.40.0-javadoc.jar.asc.sha5...  2024-07-05 22:26       128      
opentelemetry-api-1.40.0-javadoc.jar.md5          2024-07-05 22:26        32      
opentelemetry-api-1.40.0-javadoc.jar.sha1         2024-07-05 22:26        40      
opentelemetry-api-1.40.0-javadoc.jar.sha256       2024-07-05 22:26        64      
opentelemetry-api-1.40.0-javadoc.jar.sha512       2024-07-05 22:26       128      
opentelemetry-api-1.40.0-sources.jar              2024-07-05 22:26    111200      
opentelemetry-api-1.40.0-sources.jar.asc          2024-07-05 22:26       478      
opentelemetry-api-1.40.0-sources.jar.asc.md5      2024-07-05 22:26        32      
opentelemetry-api-1.40.0-sources.jar.asc.sha1     2024-07-05 22:26        40      
opentelemetry-api-1.40.0-sources.jar.asc.sha2...  2024-07-05 22:26        64      
opentelemetry-api-1.40.0-sources.jar.asc.sha5...  2024-07-05 22:26       128      
opentelemetry-api-1.40.0-sources.jar.md5          2024-07-05 22:26        32      
opentelemetry-api-1.40.0-sources.jar.sha1         2024-07-05 22:26        40      
opentelemetry-api-1.40.0-sources.jar.sha256       2024-07-05 22:26        64      
opentelemetry-api-1.40.0-sources.jar.sha512       2024-07-05 22:26       128      
opentelemetry-api-1.40.0.jar                      2024-07-05 22:26    141734      
opentelemetry-api-1.40.0.jar.asc                  2024-07-05 22:26       478      
opentelemetry-api-1.40.0.jar.asc.md5              2024-07-05 22:26        32      
opentelemetry-api-1.40.0.jar.asc.sha1             2024-07-05 22:26        40      
opentelemetry-api-1.40.0.jar.asc.sha256           2024-07-05 22:26        64      
opentelemetry-api-1.40.0.jar.asc.sha512           2024-07-05 22:26       128      
opentelemetry-api-1.40.0.jar.md5                  2024-07-05 22:26        32      
opentelemetry-api-1.40.0.jar.sha1                 2024-07-05 22:26        40      
opentelemetry-api-1.40.0.jar.sha256               2024-07-05 22:26        64      
opentelemetry-api-1.40.0.jar.sha512               2024-07-05 22:26       128      
opentelemetry-api-1.40.0.module                   2024-07-05 22:26      3948      
opentelemetry-api-1.40.0.module.asc               2024-07-05 22:26       478      
opentelemetry-api-1.40.0.module.asc.md5           2024-07-05 22:26        32      
opentelemetry-api-1.40.0.module.asc.sha1          2024-07-05 22:26        40      
opentelemetry-api-1.40.0.module.asc.sha256        2024-07-05 22:26        64      
opentelemetry-api-1.40.0.module.asc.sha512        2024-07-05 22:26       128      
opentelemetry-api-1.40.0.module.md5               2024-07-05 22:26        32      
opentelemetry-api-1.40.0.module.sha1              2024-07-05 22:26        40      
opentelemetry-api-1.40.0.module.sha256            2024-07-05 22:26        64      
opentelemetry-api-1.40.0.module.sha512            2024-07-05 22:26       128      
opentelemetry-api-1.40.0.pom                      2024-07-05 22:26      1779      
opentelemetry-api-1.40.0.pom.asc                  2024-07-05 22:26       478      
opentelemetry-api-1.40.0.pom.asc.md5              2024-07-05 22:26        32      
opentelemetry-api-1.40.0.pom.asc.sha1             2024-07-05 22:26        40      
opentelemetry-api-1.40.0.pom.asc.sha256           2024-07-05 22:26        64      
opentelemetry-api-1.40.0.pom.asc.sha512           2024-07-05 22:26       128      
opentelemetry-api-1.40.0.pom.md5                  2024-07-05 22:26        32      
opentelemetry-api-1.40.0.pom.sha1                 2024-07-05 22:26        40      
opentelemetry-api-1.40.0.pom.sha256               2024-07-05 22:26        64      
opentelemetry-api-1.40.0.pom.sha512               2024-07-05 22:26       128      
```

I assume it's useful to know which file attestation run on (or maybe it does not make sense to run it against some of them like javadoc)